### PR TITLE
port aspnet/WebSockets#149

### PR DESCRIPTION
**Pull request by anurse**
merging as soon as the builds are green because this is blocking the build.

See aspnet/WebSockets#149 for reasoning.

/cc @JunTaoLuo @moozzyk @BrennanConroy @mikaelm12 @davidfowl

### CODE REVIEW

File: test/Microsoft.AspNetCore.WebSockets.Internal.ConformanceTest/Autobahn/AutobahnExpectations.cs [hunk 1]
```diff
@@ -13,16 +13,17 @@ public class AutobahnExpectations
         private Dictionary<string, Expectation> _expectations = new Dictionary<string, Expectation>();
         public bool Ssl { get; }
         public ServerType Server { get; }
+        public string Environment { get; }
```
Comment (JunTaoLuo) [inline test/Microsoft.AspNetCore.WebSockets.Internal.ConformanceTest/Autobahn/AutobahnExpectations.cs]: Why is this added?

Comment (anurse) [inline test/Microsoft.AspNetCore.WebSockets.Internal.ConformanceTest/Autobahn/AutobahnExpectations.cs]: Good catch, it's something the WebSockets tests use but SignalR doesn't. Copy-pasta.
